nova/tests/unit/scheduler/filters/test_shard_filter.py

```diff
@@ -177,3 +177,28 @@ def test_shard_project_has_multiple_shards_per_az_resize_other_shard(self):
         self.filt_cls._PROJECT_SHARD_CACHE['foo'] = ['vc-a-0', 'vc-a-1',
                                                      'vc-b-0']
         self.assertFalse(self.filt_cls.host_passes(host, spec_obj))
+
+    def test_shard_project_has_sharding_enabled_any_host_passes(self):
+        self.filt_cls._PROJECT_SHARD_CACHE['baz'] = ['sharding_enabled']
+        aggs1 = [objects.Aggregate(id=1, name='some-az-a', hosts=['host1']),
+                 objects.Aggregate(id=1, name='vc-a-0', hosts=['host1'])]
+        aggs2 = [objects.Aggregate(id=2, name='some-az-a', hosts=['host2']),
+                 objects.Aggregate(id=2, name='vc-a-1', hosts=['host2'])]
```

Comment:
Why do we have to test a second set of aggregates?

Reply:
I wanted to test more than one, because I actually would like to test _any_, but the test might be a bit too long then...

Reply:
Yeah, I get the "any", but adding a second doesn't make this test any different unless you think `vc-a-0` might be hardcoded somewhere in the filter. We already test that both "vc-a-0" doesn't matter, as it's not in `_PROJECT_SHARD_CACHE['baz']`.

Reply:
If you think that's overkill, I can remove the second test case.

Reply:
It's not overkill, it just doesn't test anything in addition. So yeah, please remove it.